util/ttl_map.go

```diff
@@ -0,0 +1,56 @@
+package util
+
+import (
+	"sync"
+	"time"
+)
+
+type item struct {
+	value     string
+	createdAt int64
+}
+
+type TTLMap struct {
+	m map[string]*item
+	l sync.Mutex
+}
+
+func NewTTLMap(maxTTL int64, checkFrequency int64) (m *TTLMap) {
+	m = &TTLMap{m: make(map[string]*item)}
+	go func() {
```

Comment:
I think we could avoid the goroutine if we just check the ttl in Get before returning. We'd leak stale items, though (at least until someone tries to fetch them), so this works, too.

Reply:
Since we already have memory leak issues, it's probably going to be better to keep the goroutine.

Reply:
FWIW, I'm also a bit inclined to suggest simplifying this here and relying on checking the TTL when `Get` is called, and then deleting the item.

If we're concerned about memory usage we could check the TTL of all items when any item is retrieved or stored (and delete if TTL is exceeded) - whilst that increases access costs to the map slightly, it avoids a Go routine running in the background (which makes debugging harder, etc).

Another benefit of that would be that we can drop the `checkFrequency` parameter to NewTTLMap.

Reply:
Okay, I updated the PR with the simpler option. I don't myself know if `AwsDbInstanceID` is likely to change or if it's mostly constant. We can always revisit if there is a memory leak here.